MediaGalleryRenditions/Model/GetRenditionPath.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryRenditions\Model;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Filesystem\DriverInterface;
+use Magento\MediaGalleryApi\Api\Data\AssetInterface;
+use Magento\MediaGalleryRenditionsApi\Api\GetRenditionPathInterface;
+
+class GetRenditionPath implements GetRenditionPathInterface
+{
+    /**
+     * @var Filesystem\Directory\WriteInterface
+     */
+    private $directory;
+
+    /**
+     * @var DriverInterface|mixed|null
+     */
+    private $file;
+
+    private const RENDITIONS_DIRECTORY_NAME = '.renditions';
+
+    /**
+     * @param Filesystem $filesystem
+     * @param DriverInterface|null $file
+     * @throws FileSystemException
+     */
+    public function __construct(
+        Filesystem $filesystem,
+        DriverInterface $file = null
+    ) {
+        $this->directory = $filesystem->getDirectoryWrite(DirectoryList::MEDIA);
+        $this->file = $file ?: ObjectManager::getInstance()->get(Filesystem\Driver\File::class);
+    }
+
+    /**
+     * Returns Rendition image path
+     *
+     * @param AssetInterface $asset
+     * @return string
+     * @throws LocalizedException
+     */
+    public function execute(AssetInterface $asset) :string
+    {
+        return $this->getRenditionsImageDirectory($asset->getPath());
```

Comment:
I wonder if we can simplify this method implementation to
```suggestion
        if (!$this->directory->isFile($asset->getPath())) {
             throw new InvalidArgumentExcetpion(__('Incorrect asset path!'))
        }
        if (!$this->isResizable($asset)) {
             return $asset->getPath();
        }
        return $this->directory->getRelativePath(RENDITIONS_DIRECTORY_NAME . '/' . $this->directory->getRelativePath($asset->getPath()));
```
The first condition should ensure that asset path is within media directory

Reply:
I don't think we need to return the asset path if the image size is not resizeable cause GetRenditionPath should be called if the image is resizeable. Let me know your thoughts
Thanks

Reply:
I think returning the asset path if image is not resizable might simplify the client implementation

Reply:
Sorry, But I'm not able to understand how it simplify the implementation because as per the service name which is `GetRenditionPath` and it should return the rendition path but we are returning the asset path if the image is not resizeable which doesn't make sense to me. Another point if we return the asset path then we need to add an extra condition in the GenerateRenditions service to check if the asset path is not same as generate rendition path before image resizing operation and I think it will make the implementation complex. let me know your thoughts

Reply:
I believe it will be comfortable to use `GetRenditionPath` service without additional checks and conditions. The call to the service should return the path to the image that is ready to be used for the content (either from `.rendition` directory or not).
There is a check for `isResizeable` in the `GenerateRenditions` and I don't think any additional checks are needed.